src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs

```diff
@@ -181,6 +183,11 @@ private MessageType GetMessageType(ReadOnlySpan<byte> line)
         {
             EnsureStartsWithDataPrefix(line);
 
+            if (line.Length != _messagTypeLineLength)
```

Comment:
This is misplaced. GetMessageType is supposed to get the message type and not check the contents that follow the message type. I believe the check should happen here: https://github.com/aspnet/SignalR/pull/410/files#diff-b530db168b0693614299473e8c15d1d7R92 and it should just make sure that the the next two characters are \r\n. It would also remove the cryptic _messagTypeLineLength with the magic `X`.